### DISCUSSION TITLE
Automaterialization with Asset Specific Rules

### DISCUSSION THREAD

**Discussion by matthias-Q** [dagster-io/dagster#21255] [Q&A] [answered]
Hi,
we have a DAG that combines three assets into a final asset. One of the root assets is a mapping table that gets updated once in a while. See the example below:

```python
from dagster import (
    AssetExecutionContext,
    asset,
    AutoMaterializePolicy,
    AutoMaterializeRule,
)


allow_missing_parents_policy = AutoMaterializePolicy.eager().without_rules(
    AutoMaterializeRule.skip_on_parent_missing()
)


@asset(description="creates the mapping table, updated once in a blue moon")
def mapping_table(context: AssetExecutionContext):
    context.log.info("writes a static mapping table in a database")
    return {"A": "good order"}


@asset(description="fetch order data from live system")
def order_data(context: AssetExecutionContext):
    context.log.info("fetches order data from live")
    return {"user_id": 1, "order_id": 1, "order_type": "A"}


@asset(description="fetch user data from live system")
def user_data(context: AssetExecutionContext):
    context.log.info("fetches user data from live")
    return {1: "Harry"}


@asset(
    description="combined data from live with mapped values",
    auto_materialize_policy=allow_missing_parents_policy,
)
def combine_live_with_mapping(
    context: AssetExecutionContext,
    mapping_table: dict,
    user_data: dict,
    order_data: dict,
):
    user_name = user_data[order_data["user_id"]]
    order_type = mapping_table[order_data["order_type"]]

    context.log.info("creating report")
    context.log.info(
        f"user {user_name} issued a {order_type} with id {order_data['user_id']}"
    )
```
The issue now is the automaterialization policy. `combine_live_with_mapping` should run when `order_data` and `user_data` is refreshed. We don't need to wait for `mapping_table`. How can I achieve that? Using the default `AutoMaterializePolicy.eager()` issues more runs that we want.

Thanks,
Matt

**Comment by garethbrickman** [comment]
Auto-materialize policies can be customized by adding or removing rules. [More info and examples in the docs here](https://docs.dagster.io/concepts/assets/asset-auto-execution#customizing-auto-materialize-policies)

**Reply by matthias-Q** [reply]
Yes, I know, but the docs don't tell me how I can specifically mark an upstream assets as "not as important". 

So "combine_live_with_mapping" should materialize when "order_data" and "user_data" are updated but should not wait for the "mapping_table"

**Comment by OwenKephart** [comment, marked as answer]
I believe this is answered in https://github.com/dagster-io/dagster/discussions/20426, let me know if that's not the case!
